packages/owlbot-bootstrapper/common-container/mono-repo.ts

```diff
@@ -0,0 +1,144 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {execSync} from 'child_process';
+import {logger} from 'gcf-utils';
+import {Language} from './interfaces';
+import {Octokit} from '@octokit/rest';
+import {getBranchNameUtils, openABranchUtils, openAPRUtils} from './utils';
+
+/**
+ * Monorepo class
+ *
+ * @param language the Language of the particular monorepo
+ * @param repoToCloneUrl the url for the repo to clone without the preceeding https://, e.g., just github.com/googleapis/nodejs-kms.git
+ * @param repoName the name for the repo to clone, i.e., nodejs-kms
+ * @param githubToken a short-lived access Github access token
+ * @param octokit an instance of Octokit
+ */
+export class MonoRepo {
+  language: Language;
+  repoToCloneUrl: string;
+  repoName: string;
+  githubToken: string;
+  octokit: Octokit;
+
+  constructor(
+    language: Language,
+    repoToCloneUrl: string,
+    githubToken: string,
+    octokit: Octokit
+  ) {
+    this.language = language;
+    this.repoToCloneUrl = repoToCloneUrl;
+    // Get the repo name from the repoToCloneUrl, i.e. github.com/googleapis/nodejs-kms.git becomes nodejs-kms
+    this.repoName = repoToCloneUrl.split('/')[2].split('.')[0];
+    this.githubToken = githubToken;
+    this.octokit = octokit;
+  }
+
+  /**
+   * Clones a repo from github
+   *
+   * @param githubToken a short-lived access Github access token
+   * @param octokit the url for the repo to clone without the preceeding https://, e.g., just github.com/googleapis/nodejs-kms.git
+   */
+  public async _cloneRepo(
+    githubToken: string,
+    repoToCloneUrl: string,
+    directoryPath: string
+  ) {
+    try {
+      execSync(
+        `git clone https://x-access-token:${githubToken}@${repoToCloneUrl}`,
+        {cwd: directoryPath}
+      );
+    } catch (err) {
+      logger.error(err as any);
+      throw err;
+    }
+  }
+
+  /**
+   * Opens a new branch with a UUID in github in the given repo.
+   *
+   * @param repoName the name of the repository, i.e., nodejs-kms
+   */
+  public async _openABranch(repoName: string, directoryPath: string) {
+    await openABranchUtils(repoName, directoryPath);
+  }
+
+  /**
+   * Gets the name of the branch with a UUID from a well-known file path
+   */
+  public async _getBranchNameFromFile(directoryPath: string): Promise<string> {
+    return await getBranchNameUtils(directoryPath);
+  }
+
+  /**
+   * Opens a PR in github
+   *
+   * @param octokit an instantiated Octokit instance
+   * @param branchName the name of the branch with a UUID
+   * @param repoName the name of the repo to open the branch in
+   */
+  public async _openAPR(
```

Comment:
If this function simply calls `openAPRUtils()`, then why not scrap this function and call `openAPRUtils()` directly wherever `_openPR()` is called?  Then, I think renaming `openAPRUtils()` to `openAPR()` won't be potentially confusing.